src/pkg/auth/auth.go

```diff
@@ -73,6 +74,58 @@ const (
 	PromptYes Prompt = true
 )
 
+func StartAuthCodeFlowWithDocker(ctx context.Context, authPort int, tenant types.TenantName, saveToken func(string)) error {
+	redirectUri := "http://127.0.0.1:" + strconv.Itoa(authPort) + "/auth"
+
+	// Get the authorization URL before setting up the handler
+	ar, err := openAuthClient.Authorize(redirectUri, CodeResponseType, WithPkce(), WithProvider("github"))
+	if err != nil {
+		term.Error("Failed to authorize", "error", err)
+		return err
+	}
+
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/" {
+			term.Infof("Redirecting to %s", ar.url.String())
+			http.Redirect(w, r, ar.url.String(), http.StatusFound)
+			return
+		}
+		if r.URL.Path != "/auth" {
+			http.NotFound(w, r)
+			return
+		}
+		var msg string
+		query := r.URL.Query()
+		switch {
+		case query.Get("error") != "":
+			msg = "Authentication failed: " + query.Get("error_description")
+		case query.Get("state") != ar.state:
+			msg = "Authentication error: state mismatch"
+		default:
+			msg = "Authentication successful"
+			token, err := ExchangeCodeForToken(ctx, AuthCodeFlow{code: query.Get("code"), redirectUri: redirectUri, verifier: ar.verifier}, tenant, 0)
+			if err != nil {
+				term.Error("Failed to exchange code for token", "error", err)
```

Comment:
avoid stdout here (since we're in a go-routine here). If you want to log, use `slog` or your existing logging package.